tools/c7n_azure/c7n_azure/resources/key_vault_keys.py

```diff
@@ -158,3 +158,34 @@ def _process_resource_set(self, resources, event):
                 log.warning(error)
 
         return matched
+
+@KeyVaultKeys.filter_registry.register('rotation-policy')
+class KeyVaultKeyRotationFilter(Filter):
+    """Filters keyvault keys for rotation policy
+
+    :example:
+
+    Find all keyvault key that don't have rotation policy enabled
+
+    .. code-block:: yaml
+
+        policies:
+           - name: key-vault-keys-rotation-is-disabled
+             resource: azure.keyvault-keys
+             filters:
+                - type: rotation-policy
+                  state: Disabled
+    """
+    schema = type_schema('rotation-policy',required=['state'],
+                state={'type': 'string', 'enum': ['Enabled', 'Disabled']})
+
+    def process(self, resources, event=None):
+        matched = []
+        for key in resources:
+          id = KeyProperties(key_id=key['id'])
+          client = self.manager.get_client(vault_url=id.vault_url)
+          rotation = client.get_key_rotation_policy(id.name)
+          if (self.data.get('state') == 'Disabled' and not rotation.id) or \
```

Comment:
disabled / enabled seem to be using the same condition check which likely means one of them isn't valid.